### DISCUSSION TITLE
GHA- ISO Private Ephemeral Runners in K8s with image per step

### DISCUSSION THREAD

**Discussion by DrMerlin** [community/community#7122] [Actions]
The below was my [post](https://github.com/githubevents/universe2021/discussions/192) re: actions during GH Universe, and I was asked to post it again here:

Hi there- I was really hoping that GHA would announce parity for self-hosted/private runners - to allow the specification of a container image for any given step- just like public runners, and like other competing CI tools allow for with private runners.

This is the biggest limiting factor to deciding to use GHA at scale instead of other products at my company

I realize it would require a kubernetes scheduling operator- but it seems like it shouldn't be too hard to implement.

I have seen a few community approaches to fix this- but think this really needs first party support.

Is this on the road map anywhere?

The recent "ephemeral" and "webhook" changes are nice, I guess, but don't really move the needle for my company. We need image per step.

**Comment by ethomson** [comment]
Hi @DrMerlin - thanks for the feedback.  I'm looking now at how to continue to improve ephemeral runners, especially for the container build scenarios that you're describing.  I'd love to know more about your setup - first, are you using [actions-runner-controller](https://github.com/actions-runner-controller/actions-runner-controller) or have you built your own solution?
